### ISSUE TITLE
`clear-pr-merge-commit-message` is altering the PR commit title

### ISSUE THREAD

**Issue by RussKie** [refined-github/refined-github#6547]
### Description

Whenever I force squash-merge pull-requests there is a _high chance_ that the subject of a PR will get corrupted. And it looks like this wonderful extension is responsible for this - I didn't observe the issue having the extensions disabled.

There are two manifestations of the issue:
1. The subject will have the trailing `)` removed, e.g.: `My wonderful feature (#123` for github/myorg/myproduct/pull/123.
1. The subject will have the second line appended, e.g.: `My wonderful feature (#123)Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>` for github/myorg/myproduct/pull/123.

### How to replicate the issue + URL

[redacted]

### Extension version

23.4.15

### Browser(s) used

Microsoft Edge Version 112.0.1722.48 (Official build) (64-bit)

**Comment by fregante** [comment]
Thank you for the detailed bug report. 

We do have a feature that works on both of those fields when squashing: `sync-pr-commit-title`

Now I'm not sure why this is happening on force-merged PRs specifically since the UI seems to be the same.

**Comment by RussKie** [comment]
> We do have a feature that works on both of those fields when squashing: [`sync-pr-commit-title`](https://github.com/refined-github/refined-github/blob/main/source/features/sync-pr-commit-title.tsx)

Thank you for pointing me to it. I'm going to disable it and see if the issue goes away.

**Comment by fregante** [comment]
I have a guess as to why this is happening: is the PR title longer than 70 characters? That would explain the tail of the title appearing in the description box, that's a native GitHub behavior. 

Also there's a related feature that clears the description field but I don't remember its name.

**Comment by RussKie** [comment]
No, not _that_ long :) With the extension disabled the subject isn't getting trimmed or appended to.

**Comment by RussKie** [comment]
> How many commits does the PR have?

There's usually only one commit. Though I have never made any connections.

Here's a recording with the extension disabled:

https://user-images.githubusercontent.com/4403806/233517065-9c886fc8-9bcd-460f-a25c-8f275468eb50.mp4

**Comment by fregante** [comment]
I'm not seeing any issues on this repo, but also I'm not seeing `clear-pr-merge-commit-message` (that's likely due to the outdated usage of `onPrMergePanelOpen`)

<img width="726" alt="Screenshot 1" src="https://user-images.githubusercontent.com/1402241/233561197-a93b9b19-edbe-46ef-ae02-e2b0248b8163.png">


Related code that generates the PR commit title:

https://github.com/refined-github/refined-github/blob/db4e57a3f67af5374219e4deb30e1f78de35fc3c/source/features/sync-pr-commit-title.tsx#L21-L24

**Comment by fregante** [comment]
> There's usually only one commit. Though I have never made any connections.

Why use squash merges in that case?

I just checked and actually we only have that check in the feature that clears the **description**:

https://github.com/refined-github/refined-github/blob/db4e57a3f67af5374219e4deb30e1f78de35fc3c/source/features/clear-pr-merge-commit-message.tsx#L45-L48


So here's what's happening, I think unrelated to forced merges:

- `sync-pr-commit-title` is generating the wrong title 
- `clear-pr-merge-commit-message` is skipped

That means that what you have in the description box is unrelated to Refined GitHub, it's just the native behavior which is due to the long **commit** title.

**Comment by fregante** [comment]
I appreciate the detailed issue report but it's impossible to actually determine what's happening since the content is private/altered :( 

What your screencast suggests is likely not a direct consequence of Refined GitHub as it's impossible that it sets a parens-less title, it's set all at once here:

https://github.com/refined-github/refined-github/blob/f33e0da2d4b1694c642c73705b0e9b9c5085e787/source/features/sync-pr-commit-title.tsx#L23


I'll have to close this until reproducible in the wild. 
(I deleted your first screencast because the private URL was showing in some frames)

**Comment by RussKie** [comment]
> > There's usually only one commit.
> 
> Why use squash merges in that case instead of just rebase merges?

Repo policies. 

> I'll have to close this until reproducible in the wild.

No worries. Thanks for your time.

**Comment by RussKie** [comment]
Yep, that's the second issue I described above.

**Comment by fregante** [comment]
I just experienced something similar when I opened the merge form just as a new commit was pushed. This caused the form to close and, likely, `sync-pr-commit-title` to injected it into whatever field was focused: the comment field.

<img width="376" alt="Screenshot 14" src="https://user-images.githubusercontent.com/1402241/234582264-7030c446-7cd1-46ff-9215-0a452662a103.png">
